### ISSUE TITLE
Render artifacts from auto-narrowing end-of-line overhang

### ISSUE THREAD

**Issue by carlescufi** [mintty/mintty#1179]
I am seeing some render artifacts with my current setup:

- mintty 3.6.1
- [CodeNewRoman NF](https://www.nerdfonts.com/font-downloads)
- ssh+tmux

See the two "underscores" (they are just artifacts):

![image](https://user-images.githubusercontent.com/12450381/199521758-39cfcc22-4eaa-4b2d-922a-3cf9a92ea76a.png)

**Comment by mintty** [comment]
I don't reproduce such artefact. Please provide a minimal test case. I.e., without ssh and tmux if possible, and as a simple text file to be `cat`ted in order to demonstrate the issue.

**Comment by carlescufi** [comment]
@mintty let me see if I can replicate easily. Not sure which triple I am using:

![image](https://user-images.githubusercontent.com/12450381/199537185-3a67a517-44db-472b-bf0c-66a9b2fef0d9.png)

**Comment by carlescufi** [comment]
I managed to repro a very similar artifact by repeatedly invoking:

```
$ echo "                                                                          ❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯                                                                                                           ❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯❯                                "

(characters do not display properly on GitHub)

See screenshot, artifacts on right-hand-side, at the bottom:

![image](https://user-images.githubusercontent.com/12450381/199540140-e0976c5e-4ca1-4c7e-84a4-0761ab342771.png)

**Comment by mintty** [comment]
I guess your screen width is 244?
And the font file installed is Code New Roman Nerd Font Complete Windows Compatible.otf?
Anyway, still not reproduced.

**Comment by carlescufi** [comment]
> I guess your screen width is 244? 
Not sure, how do I check?

>And the font file installed is Code New Roman Nerd Font Complete Windows Compatible.otf? Anyway, still not reproduced.
I am pretty sure it is, yes.

What else would you like me to provide?

**Comment by mintty** [comment]
Check screen size with `stty size`. Check font size in Options.
Does it happen with only one file catted?
Upload test case file here as attachment, to be sure.

**Comment by carlescufi** [comment]
> Check screen size with `stty size`. 

```
$ stty size
70 253
```

> Check font size in Options. 

![image](https://user-images.githubusercontent.com/12450381/199686713-c6e09467-02c4-4fb4-a073-74c80566c29d.png)

Does it happen with only one file catted? Upload test case file here as attachment, to be sure.

It happens with a single `cat` of this file:
[artifacts.txt](https://github.com/mintty/mintty/files/9927066/artifacts.txt)

**Comment by mintty** [comment]
There are two symptoms shown in this issue: the two lower bars in your first screenshot, and the thin vertical lines at the right edge, which I can now reproduce, also at smaller screen size, e.g. 20x118.
Likely an artefact of rendering characters that appear wider than a screen cell, although mintty has already a number of mechanisms to handle that.

**Comment by carlescufi** [comment]
> There are two symptoms shown in this issue: the two lower bars in your first screenshot, and the thin vertical lines at the right edge, which I can now reproduce, also at smaller screen size, e.g. 20x118.

That's right. Hopefully they are two symptoms of the same underlying issue?

**Comment by mintty** [comment]
These artifacts do not depend on specific fonts or characters, they can be produced e.g. with
`printf "%$(($COLUMNS - 2))d ℹ\n" 0`
(DejaVu Sans Mono font).

For italic overhang, there is a mechanism to clear them out already, I hope it can be applied to this case too.

**Comment by mintty** [comment]
Released 3.7.2.
